client/signer.go

```diff
@@ -43,8 +43,11 @@ func (s *Signer) SignTransaction(tx *sqproto.UnsignedTransaction) (*sqproto.Sign
 	}, nil
 }
 
-func (s *Signer) PrivateKey() ed25519.PrivateKey {
-	return s.private
+// Seed returns the 32-byte "seed" for the key, which is used as the
+// input to generate a private key in the rust implementation, ie:
+// `ed25519_consensus::SigningKey::from(seed)`
+func (s *Signer) Seed() [ed25519.SeedSize]byte {
+	return [ed25519.SeedSize]byte(s.private.Seed())
```

Comment:
`Seed()` is already
```
func (priv PrivateKey) Seed() []byte {
	return bytes.Clone(priv[:SeedSize])
}
```

Do we need the extra byte() call?

Reply:
yeah this converts it from `[]byte` to `[32]byte` aka `[ed25519.SeedSize]byte`

Reply:
Yeah I guess after today's debacle, we want to be super specific about the sizes here!